lib/Request.php

```diff
@@ -117,6 +120,13 @@ function setAbsoluteUrl(string $url) {
      */
     function getAbsoluteUrl() : string {
 
+        if (!$this->absoluteUrl) {
+            // Guessing we're a http endpoint.
+            $this->absoluteUrl = 'http://' .
```

Comment:
Shouldnt we guess the protocoll based on e.g. The request uri, port or something?

Or maybe define the default protocol somewhere?